packages/twenty-front/src/modules/ui/input/components/SelectInput.tsx

```diff
@@ -0,0 +1,180 @@
+import { DropdownMenuItemsContainer } from '@/ui/layout/dropdown/components/DropdownMenuItemsContainer';
+import { DropdownMenuSearchInput } from '@/ui/layout/dropdown/components/DropdownMenuSearchInput';
+import { DropdownMenuSeparator } from '@/ui/layout/dropdown/components/DropdownMenuSeparator';
+import { MenuItemSelectTag } from '@/ui/navigation/menu-item/components/MenuItemSelectTag';
+import { useScopedHotkeys } from '@/ui/utilities/hotkey/hooks/useScopedHotkeys';
+import { useSetHotkeyScope } from '@/ui/utilities/hotkey/hooks/useSetHotkeyScope';
+import { useListenClickOutside } from '@/ui/utilities/pointer-event/hooks/useListenClickOutside';
+import { useTheme } from '@emotion/react';
+import {
+  ReferenceType,
+  autoUpdate,
+  flip,
+  offset,
+  size,
+  useFloating,
+} from '@floating-ui/react';
+import { useEffect, useMemo, useRef, useState } from 'react';
+import { Key } from 'ts-key-enum';
+import { TagColor, isDefined } from 'twenty-ui';
+
+const StyledRelationPickerContainer = styled.div`
+  left: -1px;
+  position: absolute;
+  top: -1px;
+  z-index: ${({ theme }) => theme.lastLayerZIndex};
+`;
+
+interface SelectInputProps {
+  onOptionSelected: (selectedOption: SelectOption) => void;
+  options: SelectOption[];
+  onCancel?: () => void;
+  defaultOption?: SelectOption;
+  parentRef?: ReferenceType | null | undefined;
+  onFilterChange?: (filteredOptions: SelectOption[]) => void;
+  onClear?: () => void;
+  clearLabel?: string;
+}
+
+export const SelectInput = ({
+  onOptionSelected,
+  onClear,
+  clearLabel,
+  options,
+  onCancel,
+  defaultOption,
+  parentRef,
+  onFilterChange,
+}: SelectInputProps) => {
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  const theme = useTheme();
+  const [searchFilter, setSearchFilter] = useState('');
+  const [selectedOption, setSelectedOption] = useState<
+    SelectOption | undefined
+  >(defaultOption);
+
+  const optionsToSelect = useMemo(
+    () =>
+      options.filter((option) => {
+        return (
+          option.value !== selectedOption?.value &&
+          option.label.toLowerCase().includes(searchFilter.toLowerCase())
+        );
+      }) || [],
+    [options, searchFilter, selectedOption?.value],
+  );
+
+  const optionsInDropDown = useMemo(
+    () =>
+      selectedOption ? [selectedOption, ...optionsToSelect] : optionsToSelect,
+    [optionsToSelect, selectedOption],
+  );
+
+  const handleOptionChange = (option: SelectOption) => {
+    setSelectedOption(option);
+    onOptionSelected(option);
+  };
+
+  const { refs, floatingStyles } = useFloating({
+    elements: { reference: parentRef },
+    strategy: 'absolute',
+    middleware: [
+      offset(() => {
+        return parseInt(theme.spacing(2), 10);
+      }),
+      flip(),
+      size(),
+    ],
+    whileElementsMounted: autoUpdate,
+    open: true,
+    placement: 'bottom-start',
+  });
+
+  const setHotkeyScope = useSetHotkeyScope();
+
+  useEffect(() => {
```

Comment:
Please implement a solution that manages this scope outside of the UI folder, Field is out of the domain of UI components